lib/kamal/secrets/adapters/bitwarden.rb

```diff
@@ -25,25 +25,31 @@ def fetch_secrets(secrets, account:, session:)
       {}.tap do |results|
         items_fields(secrets).each do |item, fields|
           item_json = run_command("get item #{item.shellescape}", session: session, raw: true)
-          raise RuntimeError, "Could not read #{secret} from Bitwarden" unless $?.success?
+          raise RuntimeError, "Could not read #{item} from Bitwarden" unless $?.success?
           item_json = JSON.parse(item_json)
-
           if fields.any?
-            fields.each do |field|
-              item_field = item_json["fields"].find { |f| f["name"] == field }
-              raise RuntimeError, "Could not find field #{field} in item #{item} in Bitwarden" unless item_field
-              value = item_field["value"]
-              results["#{item}/#{field}"] = value
-            end
+            fetch_secrets_from_fields fields, item, item_json, results
           elsif item_json.dig("login", "password")
             results[item] = item_json.dig("login", "password")
+          elsif item_json["fields"]&.any?
+            fields = item_json["fields"].pluck("name")
+            fetch_secrets_from_fields fields, item, item_json, results
           else
             raise RuntimeError, "Item #{item} is not a login type item and no fields were specified"
           end
         end
       end
     end
 
+    def fetch_secrets_from_fields(fields, item, item_json, results)
```

Comment:
Let's avoid passing the results hash into here, and return a new hash that we can merge into results. It makes it a bit clearer what's happening when calling `fetch_secrets_from_fields`

```ruby
results.merge!(fetch_secrets_from_fields(fields, item, item_json))
```